src/CosmosDB/CosmosDB/Gremlin/RestoreAzCosmosDBGremlinGraph.cs

```diff
@@ -0,0 +1,256 @@
+                    return;
+                }
+
+                Regex regex = new Regex(@"[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}");
+                var matches = regex.Matches(databaseAccount.Id);
+
+                string accountInstanceId = string.Empty;
+                if (matches.Count > 1)
+                {
+                    accountInstanceId = matches[1].Value;
+                }
+
+                DateTime latestDatabaseDeleteTime = DateTime.MinValue;
+                DateTime latestDatabaseCreateOrRecreateTime = DateTime.MinValue;
+                DateTime latestCollectionDeleteTime = DateTime.MinValue;
+                DateTime latestCollectionCreateOrRecreateTime = DateTime.MinValue;
+                string databaseRid = string.Empty;
+                string collectionRid = string.Empty;
+
+                IEnumerable restorableGremlinDatabases = CosmosDBManagementClient.RestorableGremlinDatabases.ListWithHttpMessagesAsync(databaseAccount.Location, accountInstanceId).GetAwaiter().GetResult().Body;
+                foreach (RestorableGremlinDatabaseGetResult restorableGremlinDatabase in restorableGremlinDatabases)
+                {
+                    if (restorableGremlinDatabase.Resource.OwnerId.Equals(DatabaseName))
+                    {
+                        databaseRid = restorableGremlinDatabase.Resource.OwnerResourceId;
+                        DateTime eventDateTime = DateTime.Parse(restorableGremlinDatabase.Resource.EventTimestamp);
+                        if (restorableGremlinDatabase.Resource.OperationType.Equals(OperationType.Delete) && latestDatabaseDeleteTime < eventDateTime)
+                        {
+                            latestDatabaseDeleteTime = eventDateTime;
+                        }
+
+                        if ((restorableGremlinDatabase.Resource.OperationType.Equals(OperationType.Create) || restorableGremlinDatabase.Resource.OperationType.Equals(OperationType.Recreate)) && latestDatabaseCreateOrRecreateTime < eventDateTime)
+                        {
+                            latestDatabaseCreateOrRecreateTime = eventDateTime;
+                        }
+                    }
+                }
+
+                if (latestDatabaseDeleteTime == default(DateTime))
+                {
+                    latestDatabaseDeleteTime = DateTime.MaxValue;
+                }
+
+                IEnumerable restorableGremlinGraphs = CosmosDBManagementClient.RestorableGremlinGraphs.ListWithHttpMessagesAsync(
+                    databaseAccount.Location,
+                    accountInstanceId,
+                    databaseRid,
+                    latestDatabaseCreateOrRecreateTime.ToString(),
+                    (latestDatabaseCreateOrRecreateTime < latestDatabaseDeleteTime) ? latestDatabaseDeleteTime.ToString() : DateTime.MaxValue.ToString()).GetAwaiter().GetResult().Body;
```

Comment:
DateTime.MaxValue as restore time won't work, right?
can we simplify logic as following:
for snapshot : snapshots.sorted():
    if snapshot.type == "Delete": latestDeleteSnapshot = snapshot
    else if snapshot.type == "(Re)Create": latestDeleteSnapshot = null
    
 latestDeleteSnapshot == null : not restorable